react/src/pages/ReservoirPage.tsx

```diff
@@ -823,151 +259,130 @@ const ReservoirPage: React.FC = () => {
-              type={
-                queryParams.statusCategory as 'all' | 'installed' | 'available'
+            <BAIFetchKeyButton
+              value={fetchKey}
+              autoUpdateDelay={10_000}
+              loading={isPendingTransition}
+              onChange={() =>
+                startTransition(() => {
+                  updateFetchKey();
+                  // rescanArtifacts({
+                  //   variables: {
+                  //     input: {
+                  //       storageId: 'fe878f09-06cc-4b91-9242-4c71015cce07',
+                  //       registryId: 'fe878f09-06cc-4b91-9242-4c71015cce05',
+                  //       limit: 100,
+                  //     },
+                  //   },
+                  // });
+                })
               }
-              order={queryParams.order}
-              loading={false}
-              rowSelection={{
-                type: 'checkbox',
-                preserveSelectedRowKeys: true,
-                getCheckboxProps: (record) => ({
-                  disabled: !isInstalledStatus(record.status),
-                }),
-                onChange: (selectedRowKeys) => {
-                  handleRowSelectionChange(
-                    selectedRowKeys,
-                    filteredArtifacts,
-                    setSelectedArtifactList,
-                  );
-                },
-                selectedRowKeys: _.map(selectedArtifactList, (i) => i.id),
-              }}
-              pagination={{
-                pageSize: tablePaginationOption.pageSize,
-                current: tablePaginationOption.current,
-                total: filteredArtifacts.length,
-                showTotal: (total) => (
-                  <Typography.Text type="secondary">
-                    {t('general.TotalItems', { total: total })}
-                  </Typography.Text>
-                ),
-                onChange: (current, pageSize) => {
+            />
+          </BAIFlex>
+          <BAIArtifactTable
+            artifactFragment={artifacts.edges.map((edge) => edge.node)}
+            onClickPull={(artifactId: string, revisionId: string) => {
+              artifacts.edges.forEach((artifact) => {
+                if (artifact.node.id === artifactId) {
+                  setSelectedArtifact(artifact.node);
+                  artifact.node.revisions.edges.forEach((revision) => {
+                    if (revision.node.id === revisionId) {
+                      setSelectedRevision([revision.node]);
+                      return;
+                    }
+                  });
+                  return;
+                }
+              });
+            }}
+            loading={isPendingTransition}
+            pagination={{
+              pageSize: tablePaginationOption.pageSize,
+              current: tablePaginationOption.current,
+              total: artifacts.count,
+              onChange: (current, pageSize) => {
+                startTransition(() => {
```

Comment:
You don't need to use `startTransition` because you handle it as deferredValue already.